AG-13242 - Upgrade nx to nx@20.3.1

taktran opened the pull request.

taktran committed:
AG-13242 - Upgrade nx to v20.x using migrations

taktran committed:
AG-13242 - Update patches

taktran committed:
AG-13242 - Update eslint-plugin-react-hooks

To support eslint v9

alantreadway commented on nx.json:
```diff
@@ -155,5 +171,6 @@
   },
-  "defaultProject": "all"
+  "defaultProject": "all",
+  "useLegacyCache": true
```
Should we try `false` here?

taktran replied:
Ah yes, I'll try that

alantreadway reviewed (commented):
LGTM

taktran committed:
AG-13242 - Don't use legacy cache

taktran committed:
git subrepo commit external/ag-shared

subrepo:
  subdir:   "external/ag-shared"
  merged:   "29f639f6274"
upstream:
  origin:   "https://github.com/ag-grid/ag-shared.git"
  branch:   "latest"
  commit:   "9a6c38a7c55"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-13242 - Fix up ag-shared build issues

taktran committed:
git subrepo push external/ag-shared subrepo/external/ag-shared

subrepo:
  subdir:   "external/ag-shared"
  merged:   "8ed2ee6d073"
upstream:
  origin:   "https://github.com/ag-grid/ag-shared.git"
  branch:   "latest"
  commit:   "8ed2ee6d073"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-13242 - Pin prettier to pre nx v20 upgrade version

`prettier@3.4.2` finds other issues, which we need to resolve in future
